cmd/gb-vendor/vendor/imports.go

```diff
@@ -40,3 +43,28 @@ func ParseImports(root string) (map[string]bool, error) {
 	err := filepath.Walk(root, walkFn)
 	return pkgs, err
 }
+
+// FetchMetadata fetchs the remote metadata for path.
+func FetchMetadata(path string) (io.ReadCloser, error) {
+	url := fmt.Sprintf("https://%s?go-get=1", path)
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Body, nil
+}
+
+// ParseMetadata fetchs and decodes remote metadata for path.
+func ParseMetadata(path string) (string, string, string, error) {
+	rc, err := FetchMetadata(path)
+	if err != nil {
+		return "", "", "", err
+	}
+	defer rc.Close()
+
+	meta, err := parseMetaGoImports(rc)
+	if err != nil || len(meta) < 1 {
+		return "", "", "", err
+	}
+	return meta[0].Prefix, meta[0].VCS, meta[0].RepoRoot, nil
```

Comment:
Yeah. Sorry to pull rank. I know this code sucks, but it works (at least in
the success case) so I'd like to unblock all the people waiting on
golang.org/x/ repos.



On Wed, Jun 3, 2015 at 8:17 PM, Dávid Kaya <notifications@github.com> wrote:

> In cmd/gb-vendor/vendor/imports.go
> <https://github.com/constabulary/gb/pull/183#discussion_r31609963>:
>
> > +	return resp.Body, nil
> > +}
> > +
> > +// ParseMetadata fetchs and decodes remote metadata for path.
> > +func ParseMetadata(path string) (string, string, string, error) {
> > +	rc, err := FetchMetadata(path)
> > +	if err != nil {
> > +		return "", "", "", err
> > +	}
> > +	defer rc.Close()
> > +
> > +	meta, err := parseMetaGoImports(rc)
> > +	if err != nil || len(meta) < 1 {
> > +		return "", "", "", err
> > +	}
> > +	return meta[0].Prefix, meta[0].VCS, meta[0].RepoRoot, nil
>
> Ok! It was my only comment that I've found, lgtm, you can merge it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/183/files#r31609963>.
>